File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let longest;
    let max=0;
    let len=s.length
    for(let i=0;i<len;i++){
        let x="";
        for(let j=i;j<len;j++){
            if(x.includes(s[j])){
                break;
            }
            else{
                x+=s[j];
            }
        }
        if(x.length>max){
            max=x.length;
            longest=x;
        }
    }
    return max;
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxLength=0;
        int i=0;
        int j=0;
        while(i<s.size()){
            unordered_set <char> subString={};
            while(j<s.size() && !subString.count(s[j])){
                subString.insert(s[j]);
                j++;
            }
            maxLength=max(maxLength,j-i);
            i++;
            j=i;
        }
        return maxLength;
    }
};